src/app/mod.rs

```diff
@@ -1023,6 +1023,37 @@ impl<'a, 'b> App<'a, 'b> {
         self.p.gen_completions(for_shell, out_dir.into());
     }
 
+
+    /// Generate a completions file for a specified shell at runtime.  Until `cargo install` can
+    /// install extra files like a completion script, this may be used e.g. in a command that
+    /// outputs the contents of the completion script, to be redirected into a file by the user.
+    ///
+    /// # Examples
+    ///
+    /// TODO: can this somehow reference `cli.rs` from the example above or otherwise share that
```

Comment:
It can't actually reference it, but you can link to that method, and the reader should be able to figure it out. Just tell the reader, "the cli.rs from the example above" and link:

```
[`cli.rs`]: ./struct.App.html#method.gen_completions
```